File: Top 100 Codes/01-Getting Started/20-Factor of a number/Factorofanumber.py
# Method to print the divisors of a given number
def printDivisors(n):
    print(f"The divisors of {n} are: ", end="")
    i = 1
    while i <= n:
        if n % i == 0:
            print(i, end=" ")
        i += 1
    print()  # Move to a new line after printing all divisors

# Main program to take user input and find divisors
num = int(input("Enter a number to find its divisors: "))

# Validate input to ensure it's a positive number
if num <= 0:
    print("Please enter a positive integer.")
else:
    printDivisors(num)

File: Top 100 Codes/01-Getting Started/20-Factor of a number/Factorofanumber.cpp
#include <iostream>
using namespace std;

int main() {
    int num;

    // Prompt user to enter a number
    cout << "Enter a number to find its factors: ";
    cin >> num;

    // Check if the number is positive
    if (num <= 0) {
        cout << "Please enter a positive integer." << endl;
        return 1; // Exit with an error code
    }

    // Display the factors of the number
    cout << "Factors of " << num << " are: ";
    for (int i = 1; i <= num; i++) {
        // Check if 'i' is a factor of 'num'
        if (num % i == 0)
            cout << i << " ";
    }
    cout << endl;

    return 0;
}

// Time Complexity: O(N) - Iterates from 1 to num
// Space Complexity: O(1) - No extra space is used
